Sources_v3/WebSocketClient/HealthCheckMiddleware_Tests.swift

```diff
@@ -0,0 +1,83 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+@testable import StreamChatClient
+import XCTest
+
+final class HealthCheckMiddleware_Tests: XCTestCase {
+    var middleware: HealthCheckMiddleware!
+    var webSocketClient: WebSocketClientMock!
+    
+    // MARK: - Setup
+    
+    override func setUp() {
+        super.setUp()
+        
+        webSocketClient = WebSocketClientMock()
+        middleware = HealthCheckMiddleware(webSocketClient: webSocketClient)
+    }
+    
+    override func tearDown() {
+        AssertAsync.canBeReleased(&webSocketClient)
+        
+        super.tearDown()
+    }
+    
+    // MARK: - Tests
+    
+    func test_middleware_forwardsNonHealthCheckEvents() throws {
+        let event = TestEvent()
+        
+        // Simulate incoming public event
+        let forwardedEvent = try await {
+            middleware.handle(event: event, completion: $0)
+        }
+        
+        // Assert event is forwared as it is
+        XCTAssertEqual(forwardedEvent as? TestEvent, event)
+    }
+    
+    func test_middleware_filtersHealthCheckEvents_ifClientIsDeallocated() throws {
+        let event = HealthCheckEvent(connectionId: .unique)
+        
+        // Deallocate the client
+        webSocketClient = nil
```

Comment:
You can actually use `canBeReleased()` here, too. It's even better because it will fail if there are retain cycles in the WSClient. So you then don't need to check if the client is deallocated later.

Reply:
fixed in c7b85be94bb9bec60b9efe17f7f2c4921730f77b